src/framework/ui/menu/menu.component.tsx

```diff
@@ -0,0 +1,264 @@
+ *   />
+ * );
+ *
+ * private data: MenuItemType[] = [
+ *   {
+ *     title: 'Item 1',
+ *     icon: this.Icon,
+ *   },
+ *   {
+ *     title: 'Item 2',
+ *     icon: this.Icon,
+ *   },
+ *   {
+ *     title: 'Item 3',
+ *     icon: this.Icon,
+ *   },
+ * ];
+ * ```
+ *
+ * @example With Items Groups
+ *
+ * ```
+ * // use code from example above
+ *
+ * private data: MenuItemType[] = [
+ *   { title: 'Item 1' },
+ *   {
+ *     title: 'Item 2',
+ *     subItems: [
+ *       { title: 'Item 21' },
+ *       { title: 'Item 22' },
+ *       { title: 'Item 23' },
+ *     ],
+ *   },
+ *   { title: 'Item 3' },
+ * ];
+ * ```
+ *
+ * @example Inline Styling
+ *
+ * ```
+ * // use code from example above
+ *
+ * private data: MenuItemType[] = [
+ *   { title: 'Item 1', titleStyle: { color: 'red', fontSize: 18 } },
+ *   { title: 'Item 2' },
+ *   { title: 'Item 3' },
+ * ];
+ * ```
+ */
+
+class MenuComponent extends React.Component<MenuProps> {
+
+  static styledComponentName: string = 'Menu';
+
+  private service: MenuService = new MenuService();
+
+  private onSelect = (selectedIndex: number, event: GestureResponderEvent): void => {
+    const { onSelect } = this.props;
+
+    if (onSelect) {
+      onSelect(selectedIndex, event);
+    }
+  };
+
+  private getComponentStyles = (style: StyleType): StyleType => {
+    const { dividerHeight, dividerBackgroundColor } = style;
+
+    return {
+      height: dividerHeight,
+      backgroundColor: dividerBackgroundColor,
+    };
+  };
+
+  private areThereSubItems = (item: MenuItemProps): boolean => {
+    return item.subItems && item.subItems.length !== 0;
+  };
+
+  private getIsSelected = (item: MenuItemType): boolean => {
+    const { selectedIndex } = this.props;
+
+    return selectedIndex === item.menuIndex;
+  };
+
+  private renderMenuItem = (info: ListRenderItemInfo<MenuItemProps>): MenuItemElement => {
+    const { selectedIndex, themedStyle } = this.props;
+    const separatorStyle: StyleType = this.getComponentStyles(themedStyle);
+    const isSelected: boolean = this.getIsSelected(info.item);
+
+    return this.areThereSubItems(info.item) ? (
+      <MenuGroup
+        item={info.item}
+        selectedIndex={selectedIndex}
+        separatorStyle={separatorStyle}
+        onSelect={this.onSelect}
+      />
+    ) : (
+      <MenuItem
+        {...info.item}
+        selected={isSelected}
+        onPress={this.onSelect}
+      />
+    );
+  };
+
+  private renderSeparator = (): React.ReactElement<ViewProps> => {
+    const { themedStyle } = this.props;
+    const style: StyleType = this.getComponentStyles(themedStyle);
+
+    return (
+      <View style={style}/>
+    );
+  };
+
+  public render(): React.ReactNode {
+    const { appearance, data, ...restProps } = this.props;
+    const items: MenuItemType[] = this.service.setIndexes(data);
+
+    return (
+      <List
+        ItemSeparatorComponent={this.renderSeparator}
```

Comment:
Can we re-implement it with list item borders? This will make it consistent with other components